post-publication-debias/src/main/java/eu/europeana/postpublication/debias/service/RecordAnnotationService.java

```diff
@@ -112,6 +122,28 @@ private void getProxyFieldValues(Proxy proxy, Field field, FullBean bean,  Map<S
         }
     }
 
+    public Map<String, List<String>> getValueForReferenceFields( HashMap<String, List<String>> fieldData ,FullBean bean){
+        for (Map.Entry<String, List<String>> entry : fieldData.entrySet()) {
+            if(!DebiasLanguage.isSupported(entry.getKey()) && "def".equals(entry.getKey()) ) {
```

Comment:
I'm not sure I follow this. `If the language is not supported and equals def then....`
It looks to me that we don't need the "not isSupported" part? Or am I missing something?
Also in the past we wrote similar code elsewhere that checked if a value was a reference, using a regex IIRC. This solution is probably faster, but that is assuming that uri references are always tagged as def. Did you get confirmation from Hugo Shweta that this is the case?

Reply:
The check is added  to fetch the values from pref label only if the field associated value is not available with desired language . Discussed with hugo and URI will be part of the def , the language specific entries will have the actual value and not the reference.